oauth.go

```diff
@@ -24,11 +24,34 @@ import (
 	"strings"
 	"time"
 
+	"golang.org/x/net/context"
+	_oauth2 "golang.org/x/oauth2"
+
 	"github.com/coreos/go-oidc/jose"
 	"github.com/coreos/go-oidc/oauth2"
 	"github.com/coreos/go-oidc/oidc"
 )
 
+//FIXME remove constants in the future which hopefully won't be necessary in the next releases
+const (
+	GrantTypeAuthCode     = "authorization_code"
+	GrantTypeClientCreds  = "client_credentials"
+	GrantTypeUserCreds    = "password"
+	GrantTypeImplicit     = "implicit"
+	GrantTypeRefreshToken = "refresh_token"
+
+	AuthMethodClientSecretPost  = "client_secret_post"
+	AuthMethodClientSecretBasic = "client_secret_basic"
+	AuthMethodClientSecretJWT   = "client_secret_jwt"
+	AuthMethodPrivateKeyJWT     = "private_key_jwt"
+)
+
+//FIXME remove constants in the future which hopefully won't be necessary in the next releases
```

Comment:
Is there anywhere we can import these from?

Reply:
@JoelSpeed not that I know. We can use the same approach used in `oauth2-proxy` and just use it as strings, or leave these constants and maybe refactor to another place. I'd rather have it as constants, so we have a single place to change.

Reply:
I'd also prefer constants too, was just wondering since there's a fixme whether we could get them from somewhere or if we have to have them defined, this is fine to keep though